### PR TITLE
feat(ioctl): optimize w3bstream project subcommand

### LINKED ISSUE #4012: optimize ws project subcommand
### What would you like to be added

Create or update w3bstream projects directly using local config file.

### Why is this needed

optimize interactive

### How important you think this is for IoTeX
- [ ] must have
- [ ] should have
- [x] nice to have

### PR DESCRIPTION
**Pull request by saitofun**
# Description

Fixes #4012 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by huangzhiran — APPROVED** [review]

**Review by hunshenshi — APPROVED** [review]

File: ioctl/cmd/ws/wsproject.go [hunk 1]
```diff
@@ -178,3 +197,49 @@ func getEventInputsByName(logs []*iotextypes.Log, eventName string) (map[string]
...
+	var content []byte
+	content, err = os.ReadFile(filename)
+	if err != nil {
+		err = errors.Wrap(err, errProjectConfigReadFailed.Error())
+		return
+	}
+
+	// calculate and validate hash
+	hash256b = new(hash.Hash256)
+	*hash256b = hash.Hash256b(content)
```
Comment (dustinxie) [inline ioctl/cmd/ws/wsproject.go:213]: suggest to use unnamed return `(string, *hash.Hash256, error)`
so can avoid the awkward programming here

File: ioctl/cmd/ws/wsprojectupdate.go [hunk 2]
```diff
@@ -44,32 +50,62 @@ var (
...
 
 	_ = wsProjectCreate.MarkFlagRequired("project-id")
-	_ = wsProjectCreate.MarkFlagRequired("project-uri")
-	_ = wsProjectCreate.MarkFlagRequired("project-hash")
+	_ = wsProjectCreate.MarkFlagRequired("project-config-file")
 }
 
-func updateProject(projectID uint64, uri, hash string) error {
+func updateProject(projectID uint64, filename, hashstr string) (string, error) {
+	uri, hashv, err := upload(wsProjectIPFSEndpoint, filename, hashstr)
```
Comment (dustinxie) [inline ioctl/cmd/ws/wsprojectupdate.go:61]: `createProject` used `upload`, here also use `upload`? wouldn't it overwrite existing project config?

File: ioctl/cmd/ws/wsprojectupdate.go [hunk 3]
```diff
@@ -44,32 +50,62 @@ var (
...
+	}
+	_hash, ok := inputs["hash"].([32]byte)
+	if !ok {
+		return "", errors.New("result `hash` not found in event inputs")
+	}
+	hashstr = hex.EncodeToString(_hash[:])
+
+	return fmt.Sprintf("Your project is successfully created. project id is : %d\n"+
+		"project config url:  %s\n"+
+		"project config hash: %s", _projectid, _uri, hashstr), nil
```
Comment (dustinxie) [inline ioctl/cmd/ws/wsprojectupdate.go:109]: add some tests? can skip the part upload to IPFS but test other codes

File: ioctl/cmd/ws/wsprojectupdate.go (hunk 2 above)
Comment (saitofun) [reply ioctl/cmd/ws/wsprojectupdate.go:61]: the `url` and `hash` will be written to contract and the content will store in IPFS. so the existing project would not overwrite.

**Review by dustinxie — APPROVED** [review]

**Pull request merged**